### PR TITLE
Remember the Inverter Power diagnostic

### PR DESCRIPTION
**Pull request by hufman**

### CODE REVIEW

File: lennoxs30api/s30api_async.py [hunk 1]
```diff
@@ -930,6 +932,20 @@ def processEquipments(self, message) -> bool:
...
                         self._dirty = True
                         self._dirtyList.append("single_setpoint_mode")
+                for diagnostic in equipment.get("equipment", {}).get("diagnostics", []):
+                    # look up by name instead of id
+                    # different equipments reuse IDs for different points
+                    # and this diagnostic view is less likely to be translated
+                    diagnostic_name = diagnostic.get("diagnostic", {}).get("name")
+                    diagnostic_value = diagnostic.get("diagnostic", {}).get("value")
+                    if diagnostic_name == "Inverter Input Voltage":
+                        self.diagInverterInputVoltage = float(diagnostic_value)
```
Comment (PeteRager) [inline lennoxs30api/s30api_async.py:942]: What happens if the conversion fails?

File: tests/test_diagnostic_response.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    assert lsystem.diagInverterInputVoltage == None
+    assert lsystem.diagInverterInputCurrent == None
+
+    file_path = os.path.join(script_dir, "equipments_response_energy.json")
+    with open(file_path) as f:
+        data = json.load(f)
+    api.processMessage(data)
+
+    assert lsystem.diagInverterInputVoltage == 247.0
+    assert lsystem.diagInverterInputCurrent == 4.301
```
Comment (PeteRager) [inline tests/test_diagnostic_response.py:36]: +10 on adding a unit test.  We can also test the dirty processing in here.  Take a look at the changes made in the mutation test for an example.

File: lennoxs30api/s30api_async.py (hunk 1 above)
Comment (hufman) [reply lennoxs30api/s30api_async.py]: To keep the attr_updater simple, I'll skip the conversion here and leave it to the HA sensor

File: lennoxs30api/s30api_async.py [hunk 3]
```diff
@@ -930,6 +932,20 @@ def processEquipments(self, message) -> bool:
...
+                        self._dirty = True
+                        self._dirtyList.append("diagInverterInputVoltage")
+                    if diagnostic_name == "Inverter Input Current":
+                        self.diagInverterInputCurrent = float(diagnostic_value)
+                        self._dirty = True
+                        self._dirtyList.append("diagInverterInputCurrent")
```
[the remark this answers is not in the document]
Comment (hufman) [reply lennoxs30api/s30api_async.py]: Good suggestion, this is completed

File: tests/test_diagnostic_response.py (hunk 2 above)
Comment (hufman) [reply tests/test_diagnostic_response.py]: Added some tests for failure modes and to verify some update callback behavior

**Pull request merged**